azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/VirtualMachineScaleSetNetworkInterfaceImpl.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Override
+    public String macAddress() {
+        return this.inner().macAddress();
+    }
+
+    @Override
+    public String internalDnsNameLabel() {
+        if (this.inner().dnsSettings() == null) {
+            return null;
+        }
+        return this.inner().dnsSettings().internalDnsNameLabel();
+    }
+
+    @Override
+    public String internalFqdn() {
+        if (this.inner().dnsSettings() == null) {
+            return null;
+        }
+        return this.inner().dnsSettings().internalFqdn();
+    }
+
+    @Override
+    public String internalDomainNameSuffix() {
+        if (this.inner().dnsSettings() == null) {
+            return null;
+        }
+        return this.inner().dnsSettings().internalDomainNameSuffix();
+    }
+
+    @Override
+    public List<String> dnsServers() {
+        if (this.inner().dnsSettings() == null || this.inner().dnsSettings().dnsServers() == null) {
+            return Collections.unmodifiableList(new ArrayList<String>());
+        }
+        return Collections.unmodifiableList(this.inner().dnsSettings().dnsServers());
+    }
+
+    @Override
+    public List<String> appliedDnsServers() {
+        List<String> dnsServers = new ArrayList<>();
+        if (this.inner().dnsSettings() == null || this.inner().dnsSettings().appliedDnsServers() == null) {
+            return Collections.unmodifiableList(dnsServers);
+        }
+        return Collections.unmodifiableList(this.inner().dnsSettings().appliedDnsServers());
+    }
+
+    @Override
+    public Map<String, VirtualMachineScaleSetNicIpConfiguration> ipConfigurations() {
+        List<NetworkInterfaceIPConfigurationInner> inners = this.inner().ipConfigurations();
+        if (inners == null || inners.size() == 0) {
+            return Collections.unmodifiableMap(new TreeMap<String, VirtualMachineScaleSetNicIpConfiguration>());
+        }
+        Map<String, VirtualMachineScaleSetNicIpConfiguration> nicIpConfigurations = new TreeMap<>();
+        for (NetworkInterfaceIPConfigurationInner inner : inners) {
+            VirtualMachineScaleSetNicIpConfigurationImpl nicIpConfiguration = new VirtualMachineScaleSetNicIpConfigurationImpl(inner, this, this.networkManager);
+            nicIpConfigurations.put(nicIpConfiguration.name(), nicIpConfiguration);
+        }
+        return Collections.unmodifiableMap(nicIpConfigurations);
+    }
+
+    @Override
+    public String networkSecurityGroupId() {
+        if (this.inner().networkSecurityGroup() == null) {
+            return null;
+        }
+        return this.inner().networkSecurityGroup().id();
+    }
+
+    @Override
+    public String virtualMachineId() {
+        if (this.inner().virtualMachine() == null) {
+            return null;
+        }
+        return this.inner().virtualMachine().id();
+    }
+
+    @Override
+    public Observable<VirtualMachineScaleSetNetworkInterface> createResourceAsync() {
+        // Future: We need to think of introducing ReadonlyResourceImpl
```

Comment:
i know what you mean, but perhaps as long as the user doesn't see this, we should be ok